janus.c

```diff
@@ -1027,8 +1027,8 @@ int janus_process_incoming_request(janus_request *request) {
 	} else if(!strcasecmp(message_text, "claim")) {
 		janus_mutex_lock(&session->mutex);
 		if(session->source != NULL) {
-			/* Give old transport a timeout -- is this the right thing to do? */
-			session->source->transport->session_over(session->source->instance, session->session_id, TRUE);
+			/* Notify transport that the session has been claimed */
+			session->source->transport->session_claimed(session->source->instance, session->session_id);
```

Comment:
This is partly related to the point above. My feeling is that both `session_over` and `session_claimed` here make sense, even though it might be ambiguous and confusing (what if the transport really gets rid of the session in the first callback, and involves the core? it shouldn't though). Besides, `session_over` might close the previous transport, which is not idead.

At any rate, in `session_claimed` I wouldn't put the *old* transport, but the *new* one, to tell the transport plugin which trasport instance is tied to the session *now*. If we want to avoid `session_over` closing transports in that case, we might change its signature as well to avoid it when reclaiming is involved. Does this make sense?

Reply:
I think I understand. Calling both of them seems a bit strange to me since the session isn't actually over, rather, it has just been reclaimed: a bit ambiguous as you stated. My initial thought is to pass both the old and new transport to `session_claimed` and handle all the relevant logic in that callback, but maybe I'm missing something and there is a good reason to call `session_over` as well?

Reply:
I think it's a good idea to keep them both and expand `session_over` to fix the ambiguity, mostly for the case where the old and new transports are of a different nature (e.g., you reclaimed an HTTP transport from WebSockets). In this case, a new boolean flag in `session_over` called something like `claimed` might tell the first transport plugin that yes, this is over for this transport, but only because someone claimed it, and so for this specific case the old transport should not be disconnected (for `reclaim=FALSE` the behaviour might remain the one we have). Then, a `session_reclaimed` on the second transport plugin (which might be the same as the first if the claim is done, e.g., from WebSockets to WebSockets) would tell it that the session should now be handled. In case of HTTP, this would mean:

1. session_over(old_transport, sessionId, claim=TRUE) --> remove sessionId from map
2. session_reclaimed(new_transport, sessionId) --> map sessionId to new transport

This is probably a cleaner way of handling the scenario. Adding both old and new transport to `session_reclaimed` could cause problems when switching transport protocol, as in that case the transport plugin would be passed a pointer that belongs to another plugin.